test/Ocelot.AcceptanceTests/ConfigurationMergeTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Ocelot.Configuration.File;
+using Ocelot.DependencyInjection;
+using Shouldly;
+using System;
+using System.IO;
+using TestStack.BDDfy;
+using Xunit;
+
+namespace Ocelot.AcceptanceTests;
+
+public class ConfigurationMergeTests : IDisposable
+{
+    private readonly FileConfiguration _globalConfig;
+    private readonly Steps _steps;
+
+    public ConfigurationMergeTests()
+    {
+        _steps = new Steps();
+
+        if (File.Exists(TestConfiguration.PrimaryConfigurationPath))
+        {
+            try
+            {
+                File.Delete(TestConfiguration.PrimaryConfigurationPath);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+            }
+        }
+
+        _globalConfig = new FileConfiguration
+        {
+            GlobalConfiguration = new FileGlobalConfiguration
+            {
+                RequestIdKey = "initialKey",
+            },
+        };
+    }
+
+    [Fact]
+    public void Should_run_with_config_merged_to_memory()
+    {
+        this.Given(x => _steps.GivenThereIsAConfiguration(_globalConfig, TestConfiguration.ConfigurationPartPath("global")))
+            .When(x => _steps.WhenOcelotIsRunningMergedConfig(MergeOcelotJson.ToMemory))
+            .Then(x => TheOcelotJsonFileExists(false))
```

Comment:
I suggest to show that merged config contains some important option by ensuring it exists in merged object,
It can be additional check like `.And(x => TheOptionPropertyExistsInMergedObject())`

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1227/commits/ce5e2aeb994d12fe3681391f92e3cf9d77733127 ✔️